streams/src/main/java/org/apache/kafka/streams/processor/internals/PartitionGroup.java

```diff
@@ -49,44 +46,46 @@ public TopicPartition partition() {
             return queue.partition();
         }
 
-        public RecordQueue queue() {
+        RecordQueue queue() {
             return queue;
         }
     }
 
     // since task is thread-safe, we do not need to synchronize on local variables
     private int totalBuffered;
 
-    public PartitionGroup(Map<TopicPartition, RecordQueue> partitionQueues, TimestampExtractor timestampExtractor) {
-        this.queuesByTime = new PriorityQueue<>(partitionQueues.size(), new Comparator<RecordQueue>() {
+    PartitionGroup(final Map<TopicPartition, RecordQueue> partitionQueues) {
+        queuesByTime = new PriorityQueue<>(partitionQueues.size(), new Comparator<RecordQueue>() {
 
             @Override
-            public int compare(RecordQueue queue1, RecordQueue queue2) {
-                long time1 = queue1.timestamp();
-                long time2 = queue2.timestamp();
-
-                if (time1 < time2) return -1;
-                if (time1 > time2) return 1;
+            public int compare(final RecordQueue queue1, final RecordQueue queue2) {
+                final long time1 = queue1.timestamp();
```

Comment:
> To give a concrete example, I (and all the other reviewers) missed a bug introduced in a recent Streams PR because a removed line got lost in the noise of other clean-ups.

But that's not a problem of adding `final` -- it's a problem in our process -- We started to separated commits to avoid this in the future.

And for collections, you are right. Where is my C/C++ `const` keyword? :(
(Btw: you would still use an `Unmodifyable` collection for this case)